src/groupByKey.ts

```diff
@@ -1,7 +1,19 @@
+
 type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
+export function groupByKey<T>(items:T[], key:keyof T):GroupsMap<T> {
+  const result:GroupsMap<T> = {};
+
+  items.forEach((item:T) => {
+    result[String(item[key])] = [];
+  });
+
+  items.forEach((item:T) => {
+    result[String(item[key])].push(item);
+  });
```

Comment:
It's not critical, but I think you need only one cycle for this, something like
```suggestion
  items.forEach((item: T) => {
    if (!result[String(item[key])]) {
      result[String(item[key])] = [];
    }

    result[String(item[key])].push(item);
});
```

Reply:
agree, you can do in one iteration